test/unit/change_trust_asset_test.js

```diff
@@ -0,0 +1,355 @@
+    it('returns "native" for native assets', function() {
+      const asset = StellarBase.ChangeTrustAsset.native();
+      expect(asset.getAssetType()).to.eq('native');
+    });
+
+    it('returns "credit_alphanum4" if the trustline asset code length is between 1 and 4', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'ABCD',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      expect(asset.getAssetType()).to.eq('credit_alphanum4');
+    });
+
+    it('returns "credit_alphanum12" if the trustline asset code length is between 5 and 12', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'ABCDEF',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      expect(asset.getAssetType()).to.eq('credit_alphanum12');
+    });
+
+    it('returns "liquidity_pool_shares" if the trustline asset is a liquidity pool ID', function() {
+      const asset = StellarBase.ChangeTrustAsset.liquidityPoolAsset({
+        assetA,
+        assetB,
+        fee
+      });
+      expect(asset.getAssetType()).to.eq('liquidity_pool_shares');
+    });
+  });
+
+  describe('toXDRObject()', function() {
+    it('parses a native asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.native();
+      const xdr = asset.toXDRObject();
+      expect(xdr.toXDR().toString()).to.be.equal(
+        Buffer.from([0, 0, 0, 0]).toString()
+      );
+    });
+
+    it('parses a 3-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'USD',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('USD\0');
+    });
+
+    it('parses a 4-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'BART',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('BART');
+    });
+
+    it('parses a 5-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        '12345',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum12');
+      expect(xdr.value().assetCode()).to.equal('12345\0\0\0\0\0\0\0');
```

Comment:
Where does this extra `\0` suffix come from?

Reply:
This test is already part of asset_test.js:
https://github.com/stellar/js-stellar-base/blob/6e0fa3e1a25910e193041d1f377b71f125ec4d1c/test/unit/asset_test.js#L143

I'm not sure about the motivation for it but `\0` is used to pad the asset code when converting it to XDR:
https://github.com/stellar/js-stellar-base/blob/6e0fa3e1a25910e193041d1f377b71f125ec4d1c/src/asset.js#L91